src/Rules.Framework/RulesEngine.cs

```diff
@@ -68,6 +69,31 @@ public Task<RuleOperationResult> AddRuleAsync(Rule<TContentType, TConditionType>
             return this.AddRuleInternalAsync(rule, ruleAddPriorityOption);
         }
 
+        /// <summary>
+        /// Deactivates the specified existing rule.
+        /// </summary>
+        /// <param name="rule">The rule.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException">rule</exception>
+        public Task<RuleOperationResult> DeactivateRuleAsync(Rule<TContentType, TConditionType> rule)
```

Comment:
I think that "deactivating a rule" should not consist of setting the `DateEnd` as the `DateBegin`. I think that in order to do that, we should have a boolean property on the rule named `Active` which would be set to false on this method, and the rules engine should ignore the rule after that.

This is a bit of a purist view on my end, but marking the rule as deactivated explicitly is semantically obvious, whereas a `DateEnd` equal to `DateBegin` is not necessarily interpreted as a deactivated rule.

What do you think about this? I think we should only concentrate on the support to update the `DateEnd` on this PR, and work the support to deactivate rules on another PR.

Reply:
I agree with Luis. I think it's more clear to have a property that indicates whether the rule has been deactivated or not.
Additionally, in the WebUi, we can have a fourth state "Deactivated". I'd maybe consider setting the Deactivated with the current red colour, and change the Inactive to black and grey. The reasoning for this is because the red indicates a more "severe action", which in this case is the fact that someone deliberately deactivated that rule (less severe when compared to the rule just not being active).

Reply:
![image](https://user-images.githubusercontent.com/23389613/224982400-0812c7c7-5919-4419-9b89-f415085dd75d.png)

Reply:
Can you please post the final looks?

Reply:
This is now OK on my end @carlosgoias 👌 
@jdromeiro up to you now, if you do not see any additional issues, feel free to resolve the discussion 🙂

Reply:
![image](https://github.com/Farfetch/rules-framework/assets/1099060/ef8ae1c9-394c-4e05-af8a-f8eab76d750e)
here @jdromeiro